components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorFactory.java

```diff
@@ -0,0 +1,270 @@
+            CacheManager.remove(id);
+        } else {
+            cacheStore = CacheStoreManager.get("");
+            CacheManager.remove("");
+        }
+
+        cache.setCacheStore(cacheStore);
+
+        OMAttribute collectorAttr = elem.getAttribute(ATT_COLLECTOR);
+        if (collectorAttr != null && collectorAttr.getAttributeValue() != null &&
+                "true".equals(collectorAttr.getAttributeValue())) {
+            cache.setCollector(true);
+            cache.initPattern();
+        } else {
+            cache.setCollector(false);
+
+            OMAttribute timeoutAttr = elem.getAttribute(ATT_TIMEOUT);
+            if (timeoutAttr != null && timeoutAttr.getAttributeValue() != null) {
+                cache.setTimeout(Long.parseLong(timeoutAttr.getAttributeValue().trim()));
+            } else {
+                cache.setTimeout(CachingConstants.DEFAULT_TIMEOUT);
+            }
+
+            OMAttribute maxMessageSizeAttr = elem.getAttribute(ATT_MAX_MSG_SIZE);
+            if (maxMessageSizeAttr != null && maxMessageSizeAttr.getAttributeValue() != null) {
+                cacheStore.setMaxMessageSize(Integer.parseInt(maxMessageSizeAttr.getAttributeValue().trim()));
+            }
+
+            String className = null;
+            OMElement protocolElem = elem.getFirstChildWithName(PROTOCOL_Q);
+            Map<String, Object> props = new HashMap<>();
+            if (protocolElem != null) {
+                OMAttribute typeAttr = protocolElem.getAttribute(ATT_TYPE);
+
+                if (typeAttr != null &&
+                        typeAttr.getAttributeValue() != null) {
+                    OMElement hashGeneratorElem = protocolElem.getFirstChildWithName(HASH_GENERATOR_Q);
+                    if (hashGeneratorElem != null) {
+                        className = hashGeneratorElem.getText();
+                    }
+
+                    String protocolType = typeAttr.getAttributeValue().toUpperCase().trim();
+                    cacheStore.setProtocolType(protocolType);
+                    if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(protocolType)) {
+                        OMElement methodElem = protocolElem.getFirstChildWithName(HTTP_METHODS_TO_CACHE_Q);
+                        if (methodElem != null) {
+                            String[] methods = methodElem.getText().split(",");
+                            if (!"".equals(methods[0])) {
+                                for (int i = 0; i < methods.length; i++) {
+                                    methods[i] = methods[i].toUpperCase().trim();
+                                    if (!("POST".equals(methods[i]) || "GET".equals(methods[i]) || "HEAD".equals(
+                                            methods[i]) || "PUT".equals(methods[i]) || "DELETE".equals(methods[i]) ||
+                                            "OPTIONS".equals(methods[i]) || "CONNECT".equals(methods[i]) ||
+                                            "PATCH".equals(methods[i]))) {
```

Comment:
Add HTTP methods to constants in org.wso2.carbon.mediator.cache.CachingConstants